libslide/src/bignum/add.rs

```diff
@@ -7,7 +8,14 @@ impl ops::Add for Bignum {
     type Output = Bignum;
 
     fn add(self, rhs: Bignum) -> Bignum {
-        let is_neg: bool = rhs.is_neg && self.is_neg;
+        let mut is_neg: bool = false;
+        if self.is_neg && rhs.is_neg {
+            is_neg = true;
+        } else if self.is_neg && !rhs.is_neg {
+            return rhs - abs(&self);
+        } else if !self.is_neg && rhs.is_neg {
```

Comment:
You only need to do one comparison on each branch
```suggestion
        } else if self.is_neg {
            return rhs - abs(&self);
        } else if rhs.is_neg {
```